lib/datadog/tracing.rb

```diff
@@ -0,0 +1,198 @@
+module Datadog
+  # Datadog APM tracing public API.
+  #
+  # The Datadog teams ensures that public methods in this module
+  # only receive backwards compatible changes, and breaking changes
+  # will only occur in new major versions releases.
+  # @public_api
+  module Tracing
+    class << self
+      # (see Datadog::Tracer#trace)
+      # @public_api
+      def trace(name, continue_from: nil, **span_options, &block)
+        tracer.trace(name, continue_from: continue_from, **span_options, &block)
+      end
+
+      # (see Datadog::Tracer#continue_trace!)
+      # @public_api
+      def continue_trace!(digest, &block)
+        tracer.continue_trace!(digest, &block)
+      end
```

Comment:
General reasoning is `continue_trace!` is a way of injecting state, much like you might alter an array by reference. `trace` is variable; it may alter state depending on the context under which its called.

Either way, I don't think there's a great way to make this clean-cut. Maybe `trace`/`continue_trace` is more consistent with each other, but it doesn't communicate the state-alteration. I don't think `!` on both would be appropriate.